python_modules/dagster/dagster/_utils/__init__.py

```diff
@@ -844,3 +844,13 @@ def run_with_concurrent_update_guard(
             return
         update_fn(**kwargs)
         return
+
+
+def materialize(func: Callable[..., Iterable[T]]) -> Callable[..., List[T]]:
```

Comment:
this seems like a reasonable utility but i think the name here is confusing. to me "materialize" implies that it will actually run the generator, not return a new function that runs the generator, and the term "materialize" already means something else in a dagster context and is a function in the public dagster API. Can I suggest somethig more verbose? like.... create_run_generator_callable or something?

Reply:
This is technically a decorator, so I think it makes more sense to name it more based on what it's doing rather than on the fact that it returns a function. Would `return_as_list` or something make sense?